modules/nf-core/hmmer/hmmfetch/main.nf

```diff
@@ -0,0 +1,61 @@
+process HMMER_HMMFETCH {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::hmmer=3.3.2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/hmmer:3.3.2--h87f3376_2':
+        'biocontainers/hmmer:3.3.2--h87f3376_2' }"
+
+    // The module can be called with either a key, a file containing keys or neither.
+    // In the latter case, the hmm database will be indexed and an index but no output
+    // hmm will be produced.
+    input:
+    tuple val(meta), path(hmm)
+    val   key
+    path  keyfile
+    path  index         // Only used to stage the index from a previous run
+
+    output:
+    path "selection.hmm", emit: hmm,   optional: true
+    path "*.ssi"        , emit: index, optional: true
+    path "versions.yml" , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args    = task.ext.args ?: ''
+    def keyarg  = key ?: ''
+    def kfopt   = keyfile ? '-f' : ''
+    def index   = ! key && ! keyfile ? '--index' : ''
+    def outfile = ! key && ! keyfile ? '' : '> selection.hmm'
```

Comment:
I would maybe use `prefix` here too instead of `selection` (especially since meta is now available)

Reply:
Good point.